pkg/api/client.go

```diff
@@ -3415,15 +3403,18 @@ func (c *Client) RegisterAlert(alert Alert) error {
 }
 
 // GetAlerts returns the registered alerts.
-func (c *Client) GetAlerts() (alerts []Alert, err error) {
-	resp, respErr := c.Do(http.MethodGet, alertsPath, "", nil)
+func (c *Client) GetAlerts(pageSize int) (alerts []Alert, err error) {
+	path := fmt.Sprintf("%s?pageSize=%d", alertsPath, pageSize)
+
+	var results AlertResult
+	resp, respErr := c.Do(http.MethodGet, path, "", nil)
 	if respErr != nil {
 		err = respErr
 		return
 	}
 
-	err = c.ReadJSON(resp, &alerts)
-	return
+	err = c.ReadJSON(resp, &results)
```

Comment:
What if ReadJSON() returns an error here?

Reply:
right let me fix this.